test/io/nonblock/selector.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'fiber'
+
+class Selector
+  def initialize
+    @readable = {}
+    @writable = {}
+    
+    @fiber = Fiber.current
+  end
+  
+  def run
+    while @readable.any? or @writable.any?
+      readable, writable = IO.select(@readable.keys, @writable.keys, [])
+      
+      readable.each do |io|
+        @readable[io].transfer
+      end
+      
+      writable.each do |io|
+        @writable[io].transfer
+      end
+    end
+  end
+  
+  def wait_readable(fd)
+    io = IO.for_fd(fd)
```

Comment:
I think you need `IO.for_fd(fd, autoclose: false)`, otherwise that `IO` object will close() the fd once GC'd, which is likely unintended.